crates/electrum/src/electrum_ext.rs

```diff
@@ -191,38 +186,26 @@ pub trait ElectrumExt<A> {
             .enumerate()
             .map(|(i, spk)| (i as u32, spk));
 
-        self.scan(
-            prev_tip,
-            [((), spk_iter)].into(),
-            txids,
-            outpoints,
-            usize::MAX,
-            batch_size,
-        )
+        self.sync_with_keychain([((), spk_iter)].into())
```

Comment:
This default implementation no longer makes sense The impl of sync cannot call `sync_with_keychain`.. There should be an internal function that is what `scan` used to be that both these functions call to achieve their functionality.